contracts/multisig/test/asset_proxy_owner.ts

```diff
@@ -1,511 +1,641 @@
-            }),
-            constants.AWAIT_TRANSACTION_MINED_MS,
+        assetProxyOwnerWrapper = new AssetProxyOwnerWrapper(assetProxyOwner);
+        receiver = await ContractCallReceiverContract.deployFrom0xArtifactAsync(
+            artifacts.ContractCallReceiver,
+            env.provider,
+            env.txDefaults,
+            {},
         );
     });
-    beforeEach(async () => {
-        await blockchainLifecycle.startAsync();
-    });
-    afterEach(async () => {
-        await blockchainLifecycle.revertAsync();
-    });
 
-    describe('constructor', () => {
-        it('should register passed in assetProxyContracts', async () => {
-            const assetProxyContractAddresses = [erc20Proxy.address, erc721Proxy.address];
-            const newMultiSig = await AssetProxyOwnerContract.deployFrom0xArtifactAsync(
-                artifacts.AssetProxyOwner,
-                provider,
-                txDefaults,
+    function createFunctionRegistration(
+        functionSelectorLength: number = 0,
+        destinationsLength?: number,
+        functionCallTimeLockSecondsLength?: number,
+    ): { functionSelectors: string[]; destinations: string[]; functionCallTimeLockSeconds: BigNumber[] } {
+        const _destinationsLength = destinationsLength === undefined ? functionSelectorLength : destinationsLength;
+        const _functionCallTimeLockSecondsLength =
+            functionCallTimeLockSecondsLength === undefined
+                ? functionSelectorLength
+                : functionCallTimeLockSecondsLength;
+        const functionSelectors = _.times(functionSelectorLength, () => hexRandom(4));
+        const destinations = _.times(_destinationsLength, () => hexRandom(20));
+        const functionCallTimeLockSeconds = _.times(_functionCallTimeLockSecondsLength, () =>
+            BigNumber.random() // random int > 0 and < 1000
+                .times(10000000)
+                .integerValue()
+                .mod(1000),
+        );
+        return { functionSelectors, destinations, functionCallTimeLockSeconds };
+    }
+    blockchainTests.resets('constructor', () => {
+        it('should fail if destinations.length != functionSelectors.length', async () => {
+            const reg = createFunctionRegistration(1, 2, 1);
+            const tx = TestAssetProxyOwnerContract.deployFrom0xArtifactAsync(
+                artifacts.TestAssetProxyOwner,
+                env.provider,
+                env.txDefaults,
+                artifacts,
+                reg.functionSelectors,
+                reg.destinations,
+                reg.functionCallTimeLockSeconds,
+                signerAddresses,
+                new BigNumber(REQUIRED_SIGNERS),
+                new BigNumber(DEFAULT_TIME_LOCK),
+            );
+            await expect(tx).to.revertWith(RevertReason.EqualLengthsRequired);
+        });
+        it('should fail if functionCallTimeLockSeconds.length != functionSelectors.length', async () => {
```

Comment:
We might as well add a check for `should fail if functionCallTimeLockSeconds.length != destinations.length`, and maybe also a check where all of the lengths are different. It doesn't add much for now, but I think it would be valuable for regression protection.